src/main/java/seedu/address/model/FocusMode.java

```diff
@@ -0,0 +1,54 @@
+package seedu.address.model;
+
+import static java.util.Objects.requireNonNull;
+
+import seedu.address.model.tag.Tag;
+
+/**
+ * Represents the state of the focus mode.
+ */
+public class FocusMode implements ReadOnlyFocusMode {
+
+    private boolean isActive;
+
+    //TODO: replace with class
+    private Tag focusedClass;
+
+    public FocusMode() {
+        this.isActive = false;
+    }
+
+    //TODO: replace with class
+    @Override
+    public void enter(Tag focusedClass) {
+        requireNonNull(focusedClass);
+        this.isActive = true;
+        this.focusedClass = focusedClass;
+    }
+
+    /**
+     * Exits focus mode.
+     */
+    @Override
+    public void exit() {
+        this.isActive = false;
+    }
```

Comment:
Should we consider clearing the content of the `focusedClass` variable to prevent possible bugs?